vendor/bat-native-ads/src/bat/ads/internal/security/crypto_util.cc

```diff
@@ -162,42 +120,83 @@ std::vector<uint8_t> Sha256Hash(const std::string& value) {
   return sha256;
 }
 
-bool Verify(const ConfirmationInfo& confirmation) {
-  std::string credential;
-  base::Base64Decode(confirmation.credential, &credential);
-
-  base::Optional<base::Value> value = base::JSONReader::Read(credential);
-  if (!value || !value->is_dict()) {
-    return false;
+KeyPairInfo GenerateSignKeyPairFromSeed(const std::vector<uint8_t>& seed) {
+  KeyPairInfo key_pair;
+  if (seed.empty()) {
+    return key_pair;
   }
 
-  base::DictionaryValue* dictionary = nullptr;
-  if (!value->GetAsDictionary(&dictionary)) {
-    return false;
-  }
+  std::vector<uint8_t> secret_key = seed;
+  secret_key.resize(crypto_sign_SECRETKEYBYTES);
+  std::vector<uint8_t> public_key(crypto_sign_PUBLICKEYBYTES);
+  crypto_sign_keypair(&public_key.front(), &secret_key.front(), 1);
 
-  const std::string* signature = dictionary->FindStringKey("signature");
-  if (!signature) {
-    return false;
-  }
+  key_pair.public_key = public_key;
+  key_pair.secret_key = secret_key;
 
-  VerificationSignature verification_signature =
-      VerificationSignature::decode_base64(*signature);
-  if (privacy::ExceptionOccurred()) {
-    NOTREACHED();
-    return false;
-  }
+  return key_pair;
+}
+
+KeyPairInfo GenerateBoxKeyPair() {
+  std::vector<uint8_t> public_key(crypto_box_PUBLICKEYBYTES);
+  std::vector<uint8_t> secret_key(crypto_box_SECRETKEYBYTES);
+  crypto_box_keypair(&public_key.front(), &secret_key.front());
+
+  KeyPairInfo key_pair;
+  key_pair.public_key = public_key;
+  key_pair.secret_key = secret_key;
 
-  const std::string payload = CreateConfirmationRequestDTO(confirmation);
+  return key_pair;
+}
 
-  UnblindedToken unblinded_token = confirmation.unblinded_token.value;
-  VerificationKey verification_key = unblinded_token.derive_verification_key();
-  if (privacy::ExceptionOccurred()) {
-    NOTREACHED();
-    return false;
+std::vector<uint8_t> GenerateSecretKeyFromSeed(const std::string& seed_base64) {
```

Comment:
Same question as above. The "seed" is utterly irrelevant when used with the underlying nacl primitives. See https://nacl.cr.yp.to/sign.html.

Reply:
@orspetol Please see previous comment, thanks